src/server/modules/inventory.ts

```diff
@@ -72,19 +72,31 @@ export class SellItemEvent extends ServerSocketEvent implements ServerEvent {
 export class SalvageItemEvent extends ServerSocketEvent implements ServerEvent {
   event = ServerEventName.ItemSalvage;
   description = 'Salvage an item in your inventory.';
-  args = 'itemId';
+  args = 'itemId?, itemSlot?';
 
-  async callback({ itemId } = { itemId: '' }) {
+  async callback({ itemId, itemSlot } = { itemId: '', itemSlot: '' }) {
     const player = this.player;
     if(!player) return this.notConnected();
 
-    const foundItem = player.$inventory.getItemFromInventory(itemId);
-    if(!foundItem) return this.gameError('Could not find that item in your inventory.');
+    const foundItem = (function() {
```

Comment:
please don't use anonymous functions like this. just use a normal if statement, as well as `let`.

Reply:
I wanted to keep it as `const`, but I see how that was counter-productive.